src/app/(landing)/oauth-callback/facebook/exchange-token.tsx

```diff
@@ -0,0 +1,42 @@
+"use server"
+
+import { env } from "@/env.mjs"
+import { siteConfig } from "@/config/site"
+
+export async function exchangeToken(code: string, codeVerifier: string) {
+  const url = "https://graph.facebook.com/v21.0/oauth/access_token"
```

Comment:
Whats the protocol here if facebook updates the version of the endpoint to v22 lets say?

Reply:
Read: will this break when facebook eventually pushes out an update?

Reply:
Fair point. [According to this](https://developers.facebook.com/docs/graph-api/guides/versioning/), versions are released with a 2-year rolling overlap, meaning we will have 2 years to upgrade when/if v22 is released. You can alternatively make an unversioned call, which hits the latest version no matter what. Versioned calls are the recommendation since there is a near-guarantee of no breaking changes, as long as we stay on top of major version updates and modify within the 2 year window.

Reply:
I'll pull the version number out into a constant. I think that's probably worth doing.

Reply:
dope! good info